### DISCUSSION TITLE
Hoist private key account to wagmiConfig

### DISCUSSION THREAD

**Discussion by rhlsthrm** [wevm/wagmi#3480] [General]
Redirected here from the bottom of this page: https://wagmi.sh/core/guides/viem

I do need this feature since I'm building a bot that uses private key accounts. When I use contract instance functions, I don't see a parameter to pass in an account for example: simulate does not have a way to pass in account (or `chainId` for that matter).

Either I need this feature implemented properly or a way to pass those through.

**Comment by danpopenko** [comment]
I do need this feature as well because I have developed a wallet app. Currently, it seems necessary for me to create a custom connector based on Viem account.
